extra/ldr-to-hdr.sh

```diff
@@ -0,0 +1,226 @@
+TEMP_FILEPATH="/tmp"
+TEMP_FILENAME=""
+TEMP_FILENAME_ALT="" # Used to go back-and-fourth on compiling the image.
+
+# Required files needed
+CONFIG_CALFAC_PATH=""	# Calibration Factor
+CONFIG_VIG_PATH=""	# Vignetting
+CONFIG_ND_PATH=""	# Neutral Density Filter
+CONFIG_FISHEYE_PATH=""	# Fisheye Correction
+CONFIG_CAMRESP_PATH=""	# Camera Response
+CONFIG_FISHEYE_DIAM=""	# Fisheye Diameter
+CONFIG_CROP_X=""	# Crop the image starting at X
+CONFIG_CROP_Y=""	# Crop the image starting at Y
+CONFIG_VA_V=""		# View angle of the lens vertically
+CONFIG_VA_H=""		# View angle of the lens horizontally
+CONFIG_OUT_X=""		# Output width of the image
+CONFIG_OUT_Y=""		# Output height of the image
+
+# Radiance prints the image to STDOUT, so we can use two variable to go back and fourth on applying modifiers.
+IMAGE_VAR1=""
+IMAGE_VAR2=""
+
+# Print Help function
+printhelp() {
+	echo "LDR 2 HDR"
+	echo "Automating the creation of HDR images from LDR sets"
+	echo "Outputs to CWD, with the HDR name being the MD5SUM"
+	echo "Required arguments:"
+	echo "-h [path] | HDRGen binary path"
+	echo "-r [path] | Radiance package binary path"
+	echo "-i [path/file wildcard] | Input image path/wildcard"
+	echo "-z [file] | Calibration Factoring File"
+	echo "-x [file] | Vignetting Config File"
+	echo "-c [file] | Neutral Density Filter File"
+	echo "-v [file] | Fisheye Correction File"
+	echo "-b [file] | Camera Response File"
+	echo "-n [file] | Camera Settings File"
+}
+
+# Check to see if no args were given
+
+if [[ -z $1 ]]; then
+	printhelp
+	exit 0
+fi
+
+# Argument Parsing
+
+# -h| HDRGEN path
+# -r| Radiance path
+# -i| Input images, can be regex or directory too.
+# ARUGMENTS ARE DESIGNED TO BE NEXT TO EACH OTHER ON THE KEYBOARD FOR EASE OF USE.
+# -z| Calibration Factoring File
+# -x| Vignetting Config File
+# -c| Neautral Density Filter File
+# -v| Fisheye Correction File
+# -b| Camera Response File
+# -n| Camera Settings File
+
+while getopts "h:r:i:z:x:c:v:b:n:" arg; do
+  case $arg in
+    h|r) # Setting HDRGEN path or radiance path
+      export PATH+=":$OPTARG"
+      ;;&
+    i) # Getting the input files
+      INPUT_FILES=($OPTARG*)
+      for file in "$INPUT_FILES"; do
+        if ! [[ -f "$file" ]]; then
+          echo "ERROR LOADING FILES: $file IS NOT A FILE.";
+	  echo "Other files attempting to load:"
+          printf '%s\n' "${INPUT_FILES[@]}"
+	  exit 1
+	fi
+      done;
+      ;;
+    z) # Calibration Factoring File
+      CONFIG_CALFAC_PATH=$OPTARG
+      ;;
+    x) # Vignetting Config File
+      CONFIG_VIG_PATH=$OPTARG
+      ;;
+    c) # Neutral Density Filter File
+      CONFIG_ND_PATH=$OPTARG
+      ;;
+    v) # Fisheye Correction File
+      CONFIG_FISHEYE_PATH=$OPTARG
+      ;;
+    b) # Camera Response File
+      CONFIG_CAMRESP_PATH=$OPTARG
+      ;;
+    n) # Camera Setting File
+      # Use regex magic to parse the file for all the information we need.
+      CONFIG_FISHEYE_DIAM=$(grep -Po "(?<=diameter <- ).*" $OPTARG)
```

Comment:
On mac. For me, grep complains that there is no `-P` option. I changed these to lowercase `p` since it looked like that was a valid option - not sure if it does the same thing though. I think something here is failing on my machine because it's telling me later on that I didn't input a valid camera settings file (so p may not be the same as P).

Reply:
Ah, I should of tried to see if it could work with regular regex. `-P` allows for the use of Perl regex, which I was more accustomed to using.